packages/runner/src/run.ts

```diff
@@ -132,12 +132,10 @@ export async function runTest(test: Test, runner: VitestRunner) {
 
   setCurrentTest(test)
 
-  const repeats = typeof test.repeats === 'number' ? test.repeats : 1
+  const { repeats, retry } = test
 
-  for (let repeatCount = 0; repeatCount < repeats; repeatCount++) {
-    const retry = test.retry || 1
-
-    for (let retryCount = 0; retryCount < retry; retryCount++) {
+  for (let repeatCount = 0; repeatCount <= repeats; repeatCount++) {
+    for (let retryCount = 0; retryCount <= retry; retryCount++) {
```

Comment:
There is still an issue that retry count resets with each repeat

Reply:
In my opinion, I think repeat should reset the retry count, repeat should have a separate retry count.

Reply:
Repeat does not make the test stable, when a test is bound to fail once or twice, then if he uses repeat the test will not succeed

Reply:
Repeat is not for stability, retry is. We lose all information about previous retries which is bad. We should either store it differently or increment it each time

Reply:
"increment" sounds good

Reply:
<img width="440" alt="image" src="https://github.com/vitest-dev/vitest/assets/29533304/50759208-a481-4564-a60b-686012c9fd33">

Now retryCount will increase with each repeat.